CASC/PortfolioMode.cpp

```diff
@@ -224,80 +223,116 @@ bool PortfolioMode::performStrategy(Shell::Property* property)
+      extra_opts.push("ind=struct:sik=one:indgen=on:indoct=on:drc=off");
+      extra_opts.push("ind=struct:sik=one:indgen=on");
+      extra_opts.push("ind=struct:sik=one:indoct=on");
+      extra_opts.push("ind=struct:sik=all:indmd=1");
+    }
+
+    // If in SMT-COMP mode try guessing the goal (and adding the twee trick!)
+    if(env.options->schedule() == Options::Schedule::SMTCOMP){
+      extra_opts.push("gtg=exists_all:tgt=full");
+    }
+    else
+    {
+      // Don't try this in SMT-COMP mode as it requires a goal
+      extra_opts.push("slsq=on");
+      extra_opts.push("tgt=full");
+    }
 
+    // (almost) all of the above are integrated into SNAKE schedules
+    if (env.options->schedule() == Options::Schedule::SNAKE_TPTP_UNS) 
+    {
+      extra_opts.reset();
+      extra_opts.push("rp=on");
+    } else if (env.options->schedule() == Options::Schedule::SNAKE_TPTP_SAT)
+    {
+      extra_opts.reset();
+      extra_opts.push("rp=on:fmbksg=on");
+    }
   }
 
   Schedule::BottomFirstIterator it(old);
   while(it.hasNext()){
-      vstring s = it.next();
-      // try and grab time string
-      vstring ts = s.substr(s.find_last_of("_")+1,vstring::npos);
-      int t;
-      if(Lib::Int::stringToInt(ts,t)){
-        vstring prefix = s.substr(0,s.find_last_of("_")); 
-
-        // Add a copy with increased time limit
-        vstring new_s = prefix + "_" + Lib::Int::toString(t*time_multiplier);
-        extra.push(new_s);
-
-        // Add copies with new extra options (keeping the old time limit) 
+    vstring s = it.next();
+
+    if (time_multiplier > 1.0) { // also blow up any present instruction limits      
+      size_t bidx = s.rfind(":i=");
+      if (bidx == vstring::npos) {
+        bidx = s.rfind("_i=");
+      }
+      if (bidx != vstring::npos) {
+        bidx += 3; // advance past the "*i=" bit
+        size_t eidx = s.find_first_of(":_",bidx); // find the end of the number there
+        ASS_NEQ(eidx,vstring::npos);
+        vstring instrStr = s.substr(bidx,eidx-bidx);
+        unsigned instr;
+        ALWAYS(Int::stringToUnsignedInt(instrStr,instr));
+        instr *= time_multiplier;
+        s = s.substr(0,bidx) + Lib::Int::toString(instr) + s.substr(eidx);
+      }
+    }
+
+    // try and grab time string
+    vstring ts = s.substr(s.find_last_of("_")+1,vstring::npos);
+    int t;
+    if(Lib::Int::stringToInt(ts,t)){ 
+      vstring prefix = s.substr(0,s.find_last_of("_")); 
+
+      // Add a copy with increased time limit ...
+      vstring new_time_suffix = Lib::Int::toString((int)(t*time_multiplier));
+
+      if (add_extra) {
+        // ... for every possible extra_opt
         Stack<vstring>::Iterator addit(extra_opts);
-        while(addit.hasNext()){
-          vstring new_s = prefix + ":" + addit.next() + "_" + Lib::Int::toString(t);
+        while(addit.hasNext()) {
+          vstring new_s = prefix + ((prefix.back() != '_') ? ":" : "") + addit.next() + "_" + new_time_suffix;
```

Comment:
Can I check what the prefix.back() != '_' case is?

Reply:
That for inserting into the empty main part of the strat string: "fmb+10_1:1__100"